### PR TITLE
Case-insensitive names for metric instruments

### PR DESCRIPTION
**Pull request by thisthat**
This PR properly addresses the requirements for metric instrument names.
[Spec link](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/metrics/api.md#instrument-naming-requirements)

### CODE REVIEW

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/AbstractInstrument.java [hunk 1]
```diff
@@ -88,10 +88,9 @@ public int hashCode() {
 
   abstract static class Builder<B extends AbstractInstrument.Builder<?>>
       implements Instrument.Builder {
-    /* VisibleForTesting */ static final int NAME_MAX_LENGTH = 255;
     /* VisibleForTesting */ static final String ERROR_MESSAGE_INVALID_NAME =
         "Name should be a ASCII string with a length no greater than "
-            + NAME_MAX_LENGTH
+            + StringUtils.NAME_MAX_LENGTH
```
Comment (jkwatson) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/AbstractInstrument.java:93]: while we're here, can we rename this to `METRIC_NAME_MAX_LENGTH` ?

**Review by jkwatson — APPROVED** [review]
Thanks! Just one small change requested.

**Review by anuraaga — APPROVED** [review]
Thanks!

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/AbstractInstrument.java (hunk 1 above)
Comment (thisthat) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/AbstractInstrument.java]: Done in 42c17f1

**Pull request merged**